### ISSUE TITLE
[Proposal] Keep ElasticquentResultCollection@__construct has the same signature as \Illuminate\Database\Eloquent\Collection

### ISSUE THREAD

**Issue by chuangbo** [elasticquent/Elasticquent#55]
As previous discussion #49, by implement this may fix `map`, `filter`, etc. Because `Collection` use [`new static`](https://github.com/laravel/framework/blob/v5.2.29/src/Illuminate/Support/Collection.php#L175) in [many](https://github.com/laravel/framework/blob/v5.2.29/src/Illuminate/Support/Collection.php#L505) [places](https://github.com/laravel/framework/blob/v5.2.29/src/Illuminate/Support/Collection.php#L211).

`ElasticquentResultCollection@__construct` is going to accept `$items` only, and then set elasticsearch results by `ElasticquentResultCollection@setElasticsearchResults`.

**Comment by onbjerg** [comment]
This is what we're doing now. The second parameter of the constructor in ElasticquentResultCollection is optional, and thus not used when ``map`` etc is called and a new collection is returned.

**Comment by timgws** [comment]
Yeah, can you please make sure that you are using dev-master, and try again @chuangbo ?

**Comment by chuangbo** [comment]
@timgws Yes I'm using dev-master.

laravel/framework version: `v5.2.29`
```php
\App\Product::search('test')->map(function() {})
PHP error:  Undefined index: took in /app/packages/Elasticquent/src/ElasticquentResultCollection.php on line 24
```

**Comment by timgws** [comment]
What version of Elasticsearch are you using?

**Comment by chuangbo** [comment]
Elasticsearch: 2.2.1
elasticsearch-php: 2.0.3

Just found [this fork](https://github.com/clowdy/Elasticquent/blob/improvements/src/ElasticquentResultCollection.php#L20) has implemented the similar proposal.

Do you mind me to semi-cherry-pick to here?
